lib/core/test/unit/Cardano/Wallet/Primitive/AddressDiscovery/DelegationSpec.hs

```diff
@@ -0,0 +1,556 @@
+    )
+import Cardano.Wallet.Primitive.Types.Address
+    ( Address (..) )
+import Cardano.Wallet.Primitive.Types.Coin
+    ( Coin (..) )
+import Cardano.Wallet.Primitive.Types.Hash
+    ( Hash (..) )
+import Cardano.Wallet.Primitive.Types.Tx
+    ( TxIn (..), TxOut (..) )
+import Control.Arrow
+    ( first )
+import Crypto.Hash.Utils
+    ( blake2b224 )
+import Data.Map
+    ( Map )
+import Data.Set
+    ( Set )
+import Fmt
+    ( Buildable (..)
+    , Builder
+    , blockListF
+    , blockListF'
+    , blockMapF
+    , fmt
+    , listF
+    , listF'
+    , mapF'
+    , pretty
+    )
+import GHC.Generics
+    ( Generic )
+import Test.Hspec
+    ( Spec, describe, it, shouldBe )
+import Test.QuickCheck
+    ( Arbitrary (..)
+    , NonNegative (..)
+    , Property
+    , conjoin
+    , counterexample
+    , forAllShow
+    , frequency
+    , genericShrink
+    , property
+    , sublistOf
+    , withMaxSuccess
+    , (.&&.)
+    , (===)
+    )
+import Test.QuickCheck.Arbitrary.Generic
+    ( genericArbitrary )
+
+import qualified Cardano.Wallet.Primitive.Types.TokenBundle as TB
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Char8 as B8
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+
+spec :: Spec
+spec = do
+    describe "Goldens" $ do
+        let acc = toRewardAccount @StakeKey' . toEnum
+        let regAndDeleg i = applyTx (Tx
+                [ RegisterKey $ acc i
+                , Delegate $ acc i
+                ] [] []) (error "hash not needed")
+
+        let s0 = initialDelegationState accK
+        describe "initialDelegationState" $ do
+            it "presentableKeys is [0]" $ do
+                (presentableKeys s0) `shouldBe` [toEnum 0]
+            it "usableKeys is [0]" $ do
+                usableKeys s0 `shouldBe` [toEnum 0]
+
+        let s1 = regAndDeleg 0 s0
+        describe "registering and delegating key 0" $ do
+            it "presentableKeys == [0, 1]" $ do
+                (presentableKeys s1) `shouldBe` [toEnum 0, toEnum 1]
+            it "usableKeys is still [0]" $ do
+                usableKeys s1 `shouldBe` [toEnum 0]
+
+        let s2 = regAndDeleg 1 s1
+        describe "registering and delegating key 1" $ do
+            it "presentableKeys == [0, 1, 2]" $ do
+                (presentableKeys s2) `shouldBe` [toEnum 0, toEnum 1, toEnum 2]
+            it "usableKeys == [0, 1]" $ do
+                usableKeys s2 `shouldBe` [toEnum 0, toEnum 1]
+
+        let s3 = regAndDeleg 5 s2
+        describe "Impossible gaps in stake keys (shouldn't happen unless\
+            \ someone manually constructs txs to mess with the on-chain state)" $ do
+            it "presentableKeys == [0, 1, 2] (doesn't find 5)" $ do
+                (presentableKeys s3) `shouldBe`
+                    [toEnum 0, toEnum 1, toEnum 2]
+            it "usableKeys == [0, 1]" $ do
+                usableKeys s3 `shouldBe` [toEnum 0, toEnum 1]
+
+    it "(usableKeys s) are consequtive" $ property
```

Comment:
~~consequtive~~ -> consecutive ?

Reply:
Ah! Yes, I was suspicious about my spelling, looked it up with the macOS dictionary / force click, but missed that it said "No posts for X. Did you mean Y?", so I interpreted it as correct 😅